module/zfs/mmp.c

```diff
@@ -205,80 +205,57 @@ typedef enum mmp_vdev_state_flag {
 	MMP_FAIL_WRITE_PENDING	= (1 << 1),
 } mmp_vdev_state_flag_t;
 
-static vdev_t *
-mmp_random_leaf_impl(vdev_t *vd, int *fail_mask)
-{
-	int child_idx;
-
-	if (vd->vdev_ops->vdev_op_leaf) {
-		vdev_t *ret;
-
-		if (!vdev_writeable(vd)) {
-			*fail_mask |= MMP_FAIL_NOT_WRITABLE;
-			ret = NULL;
-		} else if (vd->vdev_mmp_pending != 0) {
-			*fail_mask |= MMP_FAIL_WRITE_PENDING;
-			ret = NULL;
-		} else {
-			ret = vd;
-		}
-
-		return (ret);
-	}
-
-	if (vd->vdev_children == 0)
-		return (NULL);
-
-	child_idx = spa_get_random(vd->vdev_children);
-	for (int offset = vd->vdev_children; offset > 0; offset--) {
-		vdev_t *leaf;
-		vdev_t *child = vd->vdev_child[(child_idx + offset) %
-		    vd->vdev_children];
-
-		leaf = mmp_random_leaf_impl(child, fail_mask);
-		if (leaf)
-			return (leaf);
-	}
-
-	return (NULL);
-}
-
 /*
  * Find a leaf vdev to write an MMP block to.  It must not have an outstanding
  * mmp write (if so a new write will also likely block).  If there is no usable
- * leaf in the tree rooted at in_vd, a nonzero error value is returned, and
- * *out_vd is unchanged.
- *
- * The error value returned is a bit field.
```

Comment:
did you mean to remove this sentence?

Reply:
I just moved the sentence into the preceding paragraph (at the end).